src/config.ts

```diff
@@ -333,15 +341,15 @@ log.info("Session token duration set to %s seconds", tokenDurationSecs);
 
 // HTTPs-only fetch with optional keepalive agent
 // @see https://github.com/pagopa/io-ts-commons/blob/master/src/agent.ts#L10
-const httpsApiFetch = agent.getHttpsFetch(process.env);
+const simpleHttpsApiFetch = agent.getHttpsFetch(process.env);
 
 // Register a factory service to create PagoPA client.
 const pagoPAApiUrlProd = getRequiredENVVar("PAGOPA_API_URL_PROD");
 const pagoPAApiUrlTest = getRequiredENVVar("PAGOPA_API_URL_TEST");
 export const PAGOPA_CLIENT = new PagoPAClientFactory(
   pagoPAApiUrlProd,
   pagoPAApiUrlTest,
-  httpsApiFetch
+  simpleHttpsApiFetch
```

Comment:
As we are here, should we implement this for both cases as well? (a bit complex in this case, I see)

Reply:
Maybe we can add a constraint on URLs in this case with a `PetternString` type check
```typescript
const HttpsUrl = PatternString("^(https:\/\/).+$");
type HttpsUrl = t.TypeOf<typeof HttpsUrl>;

// Register a factory service to create PagoPA client.
const pagoPAApiUrlProd = HttpsUrl.decode(
  process.env.PAGOPA_API_URL_PROD
).getOrElseL(() => process.exit(1));
const pagoPAApiUrlTest = HttpsUrl.decode(
  process.env.PAGOPA_API_URL_PROD
).getOrElseL(() => process.exit(1));
```

Reply:
So that we enforce such url to be https only. Do we want this?

Reply:
https://github.com/pagopa/io-ts-commons/blob/master/src/url.ts#L31

Reply:
This check is evaluated at startup and prevents errors at runtime, sometimes this type of enforcement could be useful.